### PR TITLE
Change from `bin/Product` and `bin/tests/` to `artifacts/Product` and `artifacts/tests/`

### PR DESCRIPTION
**Pull request by jashook**
This change does not make us more arcadified; however, it does get us closer to using their layout.

I do not expect this to make the consolidation deadline. @ViktorHofer @jkoritzinsky @trylek ptal

### CODE REVIEW

File: eng/pipelines/templates/build-test-job.yml [hunk 1]
```diff
@@ -118,7 +118,7 @@ jobs:
     - task: PublishBuildArtifacts@1
       displayName: Publish Logs
       inputs:
-        pathtoPublish: $(coreClrRepoRoot)/bin/Logs
+        pathtoPublish: $(coreClrRepoRoot)/artifacts/Logs
```
Comment (jkoritzinsky) [inline eng/pipelines/templates/build-test-job.yml]: Arcade uses lowercase-L `artifacts/logs` instead of capital-L `Logs`.

**Review by jkoritzinsky — CHANGES_REQUESTED** [review]
`build.cmd` also needs to be updated to point to `artifacts` instead of `bin`.

**Comment by ViktorHofer** [comment]
I would like to see this being ported and merged into the runtime repo before we open it up for others. As a manual port either on Friday or me over the weekend..

**Comment by jashook** [comment]
Ported https://github.com/dotnet/runtime/pull/51

**Pull request closed without merging**